File: check_selectors.py

import requests
from bs4 import BeautifulSoup

url = "https://www.tunisianet.com.tn/301-pc-portable-tunisie"
headers = {'User-Agent': 'Mozilla/5.0'}
response = requests.get(url, headers=headers)
soup = BeautifulSoup(response.text, 'html.parser')

# Find next button
next_btn = soup.select_one('a.next')
print(f"Next Button (a.next): {next_btn}")

next_btn_alt = soup.select_one('.pagination a.next')
print(f"Next Button (.pagination a.next): {next_btn_alt}")

# Find products
products = soup.select('.product-miniature')
print(f"Products count (.product-miniature): {len(products)}")

if products:
    p = products[0]
    print(f"First product name selector (.product-title a): {p.select_one('.product-title a').get_text(strip=True)}")
    print(f"First product price selector (.price): {p.select_one('.price').get_text(strip=True)}")

File: check_selectors.js

const axios = require('axios');
const cheerio = require('cheerio');

async function check() {
    try {
        const response = await axios.get('https://www.tunisianet.com.tn/301-pc-portable-tunisie', {
            headers: { 'User-Agent': 'Mozilla/5.0' }
        });
        const $ = cheerio.load(response.data);


        console.log('Next Button (a.next):', $('a.next').attr('href'));

        const all_as = $('a');
        console.log('Total links:', all_as.length);

        const next_links = $('a:contains("")').length;
        console.log('Links with :', next_links);

        const possible_products = $('[class*="product"]');
        console.log('Elements with class containing "product":', possible_products.length);

        if (possible_products.length > 0) {
            console.log('First 5 classes with "product":');
            possible_products.slice(0, 5).each((i, el) => {
                console.log($(el).attr('class'));
            });
        }

    } catch (e) {
        console.error(e.message);
    }
}

check();
